staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -242,10 +248,128 @@ func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
 		_, err = w.Write(data)
 		return err
 	}
+	if s.options.StreamingCollectionsEncoding {
```

Comment:
are these byte-for-byte identical with variants of the the existing json encoder, like pretty-indent encoding?

Reply:
I didn't attempt to implement indents for the encoding, but I expect it could be done.

Reply:
I also don't think it worth the effort, as number of clients using this should be miniscule.

Reply:
if we want this improvement, I think it's important it not behave observably differently

Reply:
+1 to Jordan - we can't change the behavior

I think it's fine to not support this feature for pretty-prints, but for configuration we support it, we need to preserve the behavior.